packages/instrumentation-socket.io/src/socket.io.ts

```diff
@@ -0,0 +1,217 @@
+                    this._unwrap(moduleExports.Server.prototype, 'on');
+                }
+                this._wrap(moduleExports.Server.prototype, 'on', this._patchOn());
+                return moduleExports;
+            },
+            (moduleExports) => {
+                if (isWrapped(moduleExports.Server.prototype.emit)) {
+                    this._unwrap(moduleExports.Server.prototype, 'emit');
+                }
+                if (isWrapped(moduleExports.Server.prototype.on)) {
+                    this._unwrap(moduleExports.Server.prototype, 'on');
+                }
+            },
+            [broadcastOperatorInstrumentation, namespaceInstrumentation, socketInstrumentation]
+        );
+    }
+
+    private _patchOn() {
+        const self = this;
+        return (original: Function) => {
+            return function (ev: any, originalListener: Function) {
+                if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+                    return original.apply(this, arguments);
+                }
+                const wrappedListener = function () {
+                    const operation = 'on';
+                    const messageName = ev;
+
+                    const span: Span = self.tracer.startSpan(`socket.io ${operation} ${messageName}`, {
+                        kind: SpanKind.CONSUMER,
+                        attributes: {
+                            [SemanticAttributes.MESSAGING_SYSTEM]: 'socket.io',
+                            [SemanticAttributes.MESSAGING_DESTINATION]: originalListener.name,
+                            [SemanticAttributes.MESSAGING_OPERATION]: MessagingOperationValues.PROCESS,
+                        },
+                    });
+
+                    if (self._config.onHook) {
+                        safeExecuteInTheMiddle(
+                            () => self._config.onHook(span, { args: arguments }),
+                            (e) => {
+                                if (e) diag.error(`socket.io instrumentation: onHook error`, e);
+                            },
+                            true
+                        );
+                    }
+
+                    const result = context.with(setSpan(context.active(), span), () =>
+                        originalListener.apply(this, arguments)
+                    );
+                    span.end();
+                    return result;
+                };
+
+                return original.apply(this, [ev, wrappedListener]);
+            };
+        };
+    }
+
+    private _patchEmit() {
+        const self = this;
+        return (original: Function) => {
+            return function (ev: any, ...args: any[]) {
+                if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+                    return original.apply(this, arguments);
+                }
+                const operation = 'emit';
+                const messageName = ev;
+                const attributes = {
```

Comment:
should we also add `SemanticAttributes.MESSAGING_DESTINATION` attribute?

Reply:
what makes sense to use as the destination? 
we can either use the socket ids (not very useful IMO) or the namespace/rooms or a combination of them.
what do you think?

Reply:
we state that the destination kind is a topic, so the expectation is to have a matching destination attribute for which the MESSAGING_DESTIONATION_KIND refers with topic semantic in my opinion.
I think we should use the namespace/rooms to be able to tell what consumers are expected to receive this message as pub-sub.
We can chat about it more online maybe